File: crossecore-java/src/java_/BreakStatementBase.java
package java_;
import org.eclipse.emf.common.notify.Notification;
import org.eclipse.emf.common.notify.NotificationChain;
import org.eclipse.emf.common.notify.impl.NotificationImpl;
import org.eclipse.emf.ecore.InternalEObject;
import org.eclipse.emf.ecore.*;
import org.eclipse.emf.ecore.impl.ENotificationImpl;
public class BreakStatementBase 
extends StatementImpl implements BreakStatement
{
	private LabeledStatement _label;
	public LabeledStatement getLabel()
	{
		
			return _label;
		
		
	}
	public void setLabel(LabeledStatement value){
		if (value != _label) {
			NotificationChain msgs = null;
			if (_label != null){
				msgs = ((InternalEObject)_label).eInverseRemove(this, Java_PackageImpl.LABELEDSTATEMENT_USAGESINBREAKSTATEMENTS, BreakStatement.class, msgs);
			}
			if (value != null){
				msgs = ((InternalEObject)value).eInverseAdd(this, Java_PackageImpl.LABELEDSTATEMENT_USAGESINBREAKSTATEMENTS, BreakStatement.class, msgs);
			}
			msgs = basicSetLabel(value, msgs);
			if (msgs != null) {
				msgs.dispatch();
			}
		}
		else if (eNotificationRequired()){
			eNotify(new ENotificationImpl(this, NotificationImpl.SET,Java_PackageImpl.BREAKSTATEMENT_LABEL , value, value));
		}
	}

	
	@Override
	protected EClass eStaticClass() {
		return Java_PackageImpl.Literals.BREAKSTATEMENT;
	}
	
	@Override
	public NotificationChain eInverseAdd(InternalEObject otherEnd, int featureID, NotificationChain msgs) {
		switch (featureID) {
			case Java_PackageImpl.BREAKSTATEMENT_LABEL:
				if (_label != null){
					msgs = ((InternalEObject)_label).eInverseRemove(this, Java_PackageImpl.BREAKSTATEMENT_LABEL, LabeledStatement.class, msgs);
				}
				return basicSetLabel((LabeledStatement)otherEnd, msgs);
		}
		return super.eInverseAdd(otherEnd, featureID, msgs);
	}
	
	@Override
	public NotificationChain eInverseRemove(InternalEObject otherEnd, int featureID, NotificationChain msgs) {
		switch (featureID) {
			case Java_PackageImpl.BREAKSTATEMENT_LABEL:
				return basicSetLabel(null, msgs);
		}
		return super.eInverseRemove(otherEnd, featureID, msgs);
	}
	
	public NotificationChain basicSetLabel(LabeledStatement newobj, NotificationChain msgs) {
		LabeledStatement oldobj = _label;
		_label = newobj;
		if (eNotificationRequired()) {
			ENotificationImpl notification = new ENotificationImpl(this, NotificationImpl.SET, Java_PackageImpl.BREAKSTATEMENT_LABEL, oldobj, newobj);
			if (msgs == null){
				msgs = notification;
			}
			else{
				msgs.add(notification);
			}
		}
		return msgs;
	}
	
	@Override
	public Object eGet(int featureID, boolean resolve, boolean coreType) {
		switch (featureID) {
			case Java_PackageImpl.BREAKSTATEMENT_LABEL:
				return getLabel();
		}
		return super.eGet(featureID, resolve, coreType);
	}
	
	@Override
	public void eSet(int featureID, Object newValue) {
		switch (featureID) {
			case Java_PackageImpl.BREAKSTATEMENT_LABEL:
				setLabel((LabeledStatement) newValue);
				return;
		}
		super.eSet(featureID, newValue);
	}
	
	@Override
	public boolean eIsSet(int featureID) {
		switch (featureID) {
			case Java_PackageImpl.BREAKSTATEMENT_LABEL:
				return getLabel() != null; //single, volatile
		}
		return super.eIsSet(featureID);
	}
	
	
	
}

File: crossecore-typescript/java_/BreakStatementBase.ts
import {EClass} from "ecore/EClass";
import {Set} from "ecore/Set";
import {NotificationChain} from "ecore/NotificationChain";
import {ENotificationImpl} from "ecore/ENotificationImpl";
import {NotificationImpl} from "ecore/NotificationImpl";
import {InternalEObject} from "ecore/InternalEObject";
import {BasicEObjectImpl} from "ecore/BasicEObjectImpl";
import {BreakStatement} from "./BreakStatement";
import {Java_PackageLiterals} from "./Java_PackageLiterals";
import {LabeledStatement} from "./LabeledStatement";
import {StatementImpl} from "./StatementImpl";
		
			export class BreakStatementBase
			extends StatementImpl
			implements BreakStatement
			{
				private _label:LabeledStatement = null;
				get label():LabeledStatement{
				
					return this._label;
				}
				set label(value:LabeledStatement) {
					if (value != this._label) {
						let msgs:NotificationChain = null;
						if (this._label != null){
							msgs = (this._label).eInverseRemove(this, Java_PackageLiterals.LABELEDSTATEMENT_USAGESINBREAKSTATEMENTS, /*BreakStatement*/ null , msgs);
						}
						if (value != null){
							msgs = value.eInverseAdd(this, Java_PackageLiterals.LABELEDSTATEMENT_USAGESINBREAKSTATEMENTS, /*BreakStatement*/ null, msgs);
						}
						msgs = this.basicSetLabel(value, msgs);
						if (msgs != null) {
							msgs.dispatch();
						}
					}
					else if (this.eNotificationRequired()){
						this.eNotify(new ENotificationImpl(this, NotificationImpl.SET,Java_PackageLiterals.BREAKSTATEMENT_LABEL , value, value));
					}
				}

			
				public static eStaticClass:EClass;
				
				protected eStaticClass():EClass{
					
					return BreakStatementBase.eStaticClass;
				}
			
				public eInverseAdd(otherEnd:InternalEObject, featureID:number, msgs:NotificationChain): NotificationChain{
					switch (featureID) {
						case Java_PackageLiterals.BREAKSTATEMENT_LABEL:
							if (this._label != null){
								msgs = this._label.eInverseRemove(this, Java_PackageLiterals.BREAKSTATEMENT_LABEL, /*LabeledStatement*/ null, msgs);
							}
							return this.basicSetLabel(otherEnd as LabeledStatement, msgs);
					}
					//return this.eInverseAddFromStatement(otherEnd, featureID, msgs);
					return super.eInverseAdd(otherEnd, featureID, msgs);
				}
				//public eInverseAddFromBreakStatement = this.eInverseAdd;
				
				public eInverseRemove(otherEnd:InternalEObject, featureID:number, msgs:NotificationChain):NotificationChain{
					switch (featureID) {
						case Java_PackageLiterals.BREAKSTATEMENT_LABEL:
							return this.basicSetLabel(null, msgs);
					}
					//return this.eInverseRemoveFromStatement(otherEnd, featureID, msgs);
					return super.eInverseRemove(otherEnd, featureID, msgs);
				}
				
				//public eInverseRemoveFromBreakStatement = this.eInverseRemove;
			
				public basicSetLabel(newobj:LabeledStatement, msgs:NotificationChain):NotificationChain {
					let oldobj = this._label;
					this._label = newobj;
					if (this.eNotificationRequired()) {
						let notification = new ENotificationImpl(this, NotificationImpl.SET, Java_PackageLiterals.BREAKSTATEMENT_LABEL, oldobj, newobj);
						if (msgs == null){
							msgs = notification;
						}
						else{
							msgs.add(notification);
						}
					}
					return msgs;
				}
				
			
				public eGet_number_boolean_boolean(featureID:number, resolve:boolean, coreType:boolean):any{
					switch (featureID) {
						case Java_PackageLiterals.BREAKSTATEMENT_COMMENTS:
							return this.comments;
						case Java_PackageLiterals.BREAKSTATEMENT_ORIGINALCOMPILATIONUNIT:
							return this.originalCompilationUnit;
						case Java_PackageLiterals.BREAKSTATEMENT_ORIGINALCLASSFILE:
							return this.originalClassFile;
						case Java_PackageLiterals.BREAKSTATEMENT_LABEL:
							return this.label;
					}
					//return this.eGetFromStatement(featureID, resolve, coreType);
					return super.eGet(featureID, resolve, coreType);
				}
				
				public eSet_number_any(featureID:number, newValue:any):void {
					switch (featureID) {
						case Java_PackageLiterals.BREAKSTATEMENT_LABEL:
							this.label = <LabeledStatement> newValue;
							return;
					}
					super.eSet_number_any(featureID, newValue);
				}

				
			}
			
